### PR TITLE
feat: refactor assets path AR [AR-1578]

### PR DESCRIPTION
**Pull request by gongracr**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1578" title="AR-1578" target="_blank">TaskAR-1578</a>  Store assets as File paths instead of raw byte arrays in DB
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

With the new update in Kalium, bringing `Okio` library onboard and adding the concept of `Path`, `source` and `sink`, we also need to adapt these changes in the AR side of things. It also refactors many of the methods inside `FileUtils` to make sure that these ones take the newly defined cache and assets storage paths.

### Dependencies

This PR relies on [this one](https://github.com/wireapp/kalium/pull/649) raised in Kalium repo:

- https://github.com/wireapp/kalium/pull/649

#### Test Coverage

- [x] I have added automated test to this contribution

#### How to Test

Try to send any kind of attachment!

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/7/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/delete/MessageDeletion.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.wire.android.ui.home.conversations.delete
+
+data class MessageDeletion(val messageToDeleteId: String, val isSelfMessage: Boolean) {
+    override fun toString(): String = "$messageToDeleteId:$isSelfMessage"
```
[the remark this answers is not in the document]
Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/delete/MessageDeletion.kt:4]: I think this was a leftover from previous deletion feature, not really sure how it slipped into this PR 😓 removing in any case

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/9/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/12/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 2]
```diff
@@ -308,11 +322,13 @@ class ConversationViewModel @Inject constructor(
     }
 
     fun onSnackbarMessage(msgCode: ConversationSnackbarMessages) {
-        viewModelScope.launch {
```
[the remark this answers is not in the document]
Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:311]: This was to ensure that the UI messages get delivered on the main thread

[the remark this answers is not in the document]
Comment (typfel) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:311]: Ok

**Comment by AndroidBob** [comment]
Build [17](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/17/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3936342)(3936342)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3936342)(3936342)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/avatarpicker/AvatarPickerViewModel.kt [hunk 3]
```diff
@@ -63,21 +68,22 @@ class AvatarPickerViewModel @Inject constructor(
...
                 pictureState = avatarImageManager
                     .postProcessAvatar(imageUri)
-                    ?.let { PictureState.Picked(it) } ?: PictureState.Empty
+                    ?.let { PictureState.Picked(it) } ?: PictureState.Picked(imageUri)
             }
         }
     }
 
-    fun uploadNewPickedAvatarAndBack() {
+    fun uploadNewPickedAvatarAndBack(context: Context) {
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/avatarpicker/AvatarPickerViewModel.kt:76]: From my side it's ok to pass the `Context` here, not the best option in terms of testability, but we can improve it later 💪

File: app/src/test/kotlin/com/wire/android/framework/FakeKaliumFileSystem.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import kotlinx.coroutines.withContext
+import okio.Path
+import okio.Path.Companion.toPath
+import okio.Sink
+import okio.Source
+import okio.buffer
+import okio.fakefilesystem.FakeFileSystem
+import okio.use
+
+class FakeKaliumFileSystem(
```
Comment (yamilmedina) [inline app/src/test/kotlin/com/wire/android/framework/FakeKaliumFileSystem.kt:17]: ❤️

Comment (yamilmedina) [reply app/src/test/kotlin/com/wire/android/framework/FakeKaliumFileSystem.kt:17]: I'm happy to see Fakes as well, it's important and nice we understand the differences from other test patterns in the team 💪

**Review by yamilmedina — APPROVED** [review]
Added some small comments, but I see this is a good final touchpoint to have this improvement on Assets performance. ❤️ 🥇

**Comment by AndroidBob** [comment]
Build [19](https://10.10.124.134/job/wire-android-reloaded/job/PR-743/19/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3941808)(3941808)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3941808)(3941808)-dev-release.apk)

**Pull request merged**